pkg/crypto/bls/bbs12381g2pub/signature.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package bbs12381g2pub
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/phoreproject/bls"
+	"github.com/phoreproject/bls/g2pubs"
+)
+
+// Signature defines BLS signature.
+type Signature struct {
+	Signature *g2pubs.Signature
+
+	E *bls.FR
+	S *bls.FR
+}
+
+// ParseSignature parses a Signature from bytes.
+func ParseSignature(sigBytes []byte) (*Signature, error) {
+	if len(sigBytes) != bls12381SignatureLen {
+		return nil, errors.New("invalid size of signature")
+	}
+
+	var sigBytesArr [g1CompressedSize]byte
+
+	copy(sigBytesArr[:], sigBytes[:g1CompressedSize])
+
+	signature, err := g2pubs.DeserializeSignature(sigBytesArr)
```

Comment:
Does `DeserializeSignature` check if the point is not at infinity?

Reply:
Yes, it checks (the same as for `DeserializePublicKey()`).